File: lc/lc/2021_target/easy/golang/lc_1_two_sum.go
package main
import "fmt"


func twoSum(nums []int, target int) []int {
    m := map[int]int{}

    for index, v := range nums {
        if _, ok := m[target - v]; ok {
            return []int{index, m[target-v]}
        }
        m[v] = index
    }
    return []int{}
}

func main() {
    nums := []int { 2, 7, 11, 15}
    result := twoSum(nums, 9)
    fmt.Printf("\n Result [%v]", result)
}

File: lc/lc/2021_target/array/lc_1_two_sum.cpp
#include<iostream>
#include<vector>
#include<unordered_map>

using namespace std;


vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int,int> m;
    for ( int i = 0; i < nums.size();i++) {
       auto complement = target - nums[i];
       if (m.count(complement)) {
           return {i, m[complement]};
       }
       m[nums[i]] = i;
    }
    return {};
}

int main() {
    vector<int> nums = { 2, 7, 11, 15};
    vector<int> result = twoSum(nums, 9);
    for ( auto r : result) {
        std::cout << r << " ";
    }
    return 0;
}
